### ISSUE TITLE
[RESEARCH] BFS/SSSP with multiple starting vertices - framework 

### ISSUE THREAD

**Issue by BradReesWork** [rapidsai/cugraph#1339]
As we move to using stream to support parallel execution, there are a number of items that need to be researched.

Assume a user wants to run SSSP, or BFS, on a collection of starting vertices (SG or MNMG but not batch), the questions are:

* How many stream can be run simultaneously?  
    - how does GPU loading effect that (i.e. larger graphs, more cuDF data)?

* How to manage return data?

**Comment by afender** [comment]
Short update

> How many stream can be run simultaneously?

From what I could see while analyzing egoNet, concurrency is limited by the number of available blocks on the device. Thrust-based codes may request a lot of blocks without exposing a way to control this. A workflow with many small graphs typically leverages concurrency better than a large graph with many small tasks (as the latter may end up waiting for available resources more often).

> How to manage return data?

Planing on writing csv for each seed or group of seeds. 
As a start, I'll be investigating async host write from c++ level. Host because we want the device to continue focusing on traversals for other seeds. Inside the C++ layer because we expect one entry point for the multi-seed BFS, so writing "as we go" should happen at that level, and performance is another motivation.

**Comment by afender** [comment]
Closing. Documents are available on the team share.

**Comment by mughetto** [comment]
> Documents are available on the team share.

Sorry but what does this means in practice, did it go to backlog or?

**Comment by afender** [comment]
Hi @mughetto, glad you asked. Traversals with multiple sources are among the features we are and will continue focusing on in the coming releases. Closing this research issue just means that the research questions above have been addressed. During this release, we enabled streams in cuGraph, analyzed multi seeds pipelines, and figured out the path forward (captured in other github issues listed below). 
The next steps are: 
- one seed per component version in 0.20 #1487
- SG generic MS BFS with aggregated frontier traversal after 0.20  #1465 
Hope that helps!
